build/build.c

```diff
@@ -213,24 +278,58 @@ static rpmRC buildSpec(BTA_t buildArgs, rpmSpec spec, int what)
     } else {
 	int didBuild = (what & (RPMBUILD_PREP|RPMBUILD_BUILD|RPMBUILD_INSTALL));
 
+	if (!spec->buildrequires && (what & RPMBUILD_PACKAGESOURCE) &&
+	    !(what & (RPMBUILD_BUILD|RPMBUILD_INSTALL|RPMBUILD_PACKAGEBINARY))){
```

Comment:
This would far easier to read if at least part of it was split into a helper variable, similar to didBuild above.